### PR TITLE
Update ADAggregatedDispatcher to output only designated properties

### PR DESCRIPTION
**Pull request by jasoncoolmax**
According to the design changes, ADAggregatedDispatcher has been updated to only output some designated properties.

Since each different telemetry event (like UIEvent, ApiEvent, etc.) has different designated properties to output, a function called 
` (void)processEvent:(NSMutableDictionary*)eventToBeDispatched`
is added to each telemetry event class. Such a function will add the properties needed to be output to `eventToBeDispatched`.

Besides, 
- new telemetry properties have been added;
- telemetry-related strings have been changed to constants.

### CODE REVIEW

File: ADAL/src/telemetry/ADTelemetryEventStrings.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+extern NSString *const TELEMETRY_AUTHORITY_TYPE;
```
Comment (RPangrle) [inline ADAL/src/telemetry/ADTelemetryEventStrings.h]: Prepend AD_ to the front of all of these, as they're in a global namespace and likely to cause conflict for someone.

Comment (jasoncoolmax) [inline ADAL/src/telemetry/ADTelemetryEventStrings.h]: yes sir :)

File: ADAL/src/telemetry/ADTelemetryAPIEvent.m [hunk 2]
```diff
@@ -30,84 +33,152 @@ - (void)setResultStatus:(ADAuthenticationResultStatus)status
...
+            promptBehaviorString = @"AD_FORCE_PROMPT";
+            break;
+        default:
+            promptBehaviorString = AD_TELEMETRY_UNKNOWN;
+    }
+    
+    [self setProperty:AD_TELEMETRY_PROMPT_BEHAVIOR value:promptBehaviorString];
+}
+
+- (void)processEvent:(NSMutableDictionary*)eventToBeDispatched
```
Comment (RPangrle) [inline ADAL/src/telemetry/ADTelemetryAPIEvent.m]: It's extremely unclear to me what exactly this method is (and its superclass implementation) are trying to accomplish. Please add some comments explaining their purpose and the expected result.

File: ADAL/src/telemetry/ADTelemetryHttpEvent.m [hunk 3]
```diff
@@ -22,42 +22,80 @@
...
+    if ([eventToBeDispatched objectForKey:AD_TELEMETRY_HTTP_EVENT_COUNT])
+    {
+        httpEventCount = [[eventToBeDispatched objectForKey:AD_TELEMETRY_HTTP_EVENT_COUNT] intValue] + 1;
+    }
+    [eventToBeDispatched setObject:[NSString stringWithFormat:@"%d", httpEventCount] forKey:AD_TELEMETRY_HTTP_EVENT_COUNT];
+    
+    NSArray* properties = [self getProperties];
+    for (NSArray* property in properties)
+    {
+        if ([property[0] isEqualToString:AD_TELEMETRY_HTTP_RESPONSE_CODE]
```
Comment (RPangrle) [inline ADAL/src/telemetry/ADTelemetryHttpEvent.m]: You need to use a custom class here, not an array for this. It makes the code a lot harder to follow and understand what it is trying to accomplish.

File: ADAL/src/telemetry/ADTelemetryDefaultEvent.m [hunk 4]
```diff
@@ -130,6 +131,14 @@ - (NSString*)getStringFromDate:(NSDate*)date
     return [s_dateFormatter stringFromDate:date];
 }
 
+- (void)processEvent:(NSMutableDictionary*)eventToBeDispatched
```
Comment (RPangrle) [inline ADAL/src/telemetry/ADTelemetryDefaultEvent.m]: processEvent: is too generic of a name and doesn't accurately reflect what it's doing.

File: ADAL/src/telemetry/ADTelemetryHttpEvent.m [hunk 5]
```diff
@@ -22,42 +22,80 @@
...
+            ||[property[0] isEqualToString:AD_TELEMETRY_OAUTH_ERROR_CODE]
+            ||[property[0] isEqualToString:AD_TELEMETRY_HTTP_ERROR_DOMAIN]
+            ||[property[0] isEqualToString:AD_TELEMETRY_HTTP_PATH]
+            ||[property[0] isEqualToString:AD_TELEMETRY_HTTP_REQUEST_ID_HEADER])
+        {
+            [eventToBeDispatched setObject:property[1] forKey:property[0]];
```
Comment (RPangrle) [inline ADAL/src/telemetry/ADTelemetryHttpEvent.m]: This isn't really aggregating anything. It's collapsing and overriding data. Is this really what was spec'd?

**Comment by jasoncoolmax** [comment]
now target at dev branch

File: ADAL/src/telemetry/ADTelemetryAPIEvent.m [hunk 6]
```diff
@@ -152,31 +153,31 @@ - (void)setPromptBehavior:(ADPromptBehavior)promptBehavior
...
+            ||[property.name isEqualToString:AD_TELEMETRY_TENANT_ID]
+            ||[property.name isEqualToString:AD_TELEMETRY_USER_ID]
+            ||[property.name isEqualToString:AD_TELEMETRY_RESPONSE_TIME]
+            ||[property.name isEqualToString:AD_TELEMETRY_CLIENT_ID]
+            ||[property.name isEqualToString:AD_TELEMETRY_API_ID]
+            ||[property.name isEqualToString:AD_TELEMETRY_USER_CANCEL]
+            ||[property.name isEqualToString:AD_TELEMETRY_ERROR_CODE]
+            ||[property.name isEqualToString:AD_TELEMETRY_ERROR_DOMAIN]
+            ||[property.name isEqualToString:AD_TELEMETRY_PROTOCOL_CODE]
+            ||[property.name isEqualToString:AD_TELEMETRY_ERROR_DESCRIPTION])
```
Comment (RPangrle) [inline ADAL/src/telemetry/ADTelemetryAPIEvent.m]: Much better! 👍

**Comment by RPangrle** [comment]
I'm not seeing any tests on Telemetry where it is testing the telemetry output during actual ADAL operations. I've filed an issue to fix this later: #832

**Comment by RPangrle** [comment]
:shipit:

**Pull request merged**